crates/store/re_sorbet/src/migration.rs

```diff
@@ -231,9 +301,20 @@ pub fn reorder_columns(batch: &ArrowRecordBatch) -> ArrowRecordBatch {
 
 /// Move indicator component to archetype field name.
 // TODO(#8129): For now, this renames the indicator column metadata. Eventually, we want to remove the column altogether.
+#[tracing::instrument(level = "trace", skip_all)]
 pub fn rewire_indicator_components(batch: &ArrowRecordBatch) -> ArrowRecordBatch {
     re_tracing::profile_function!();
 
+    let needs_rewiring = batch.schema_ref().fields().iter().any(|field| {
+        field
+            .metadata()
+            .get("rerun.component")
+            .is_some_and(|value| value.ends_with("Indicator"))
```

Comment:
The decision on when to run the migration code is a bit more involved.

You also need to check if the metadata does not contain `rerun.archetype` and `rerun.archetype_field`.

https://github.com/rerun-io/rerun/blob/main/crates/store/re_sorbet/src/migration.rs#L258-L260

Also, this will change again with https://github.com/rerun-io/rerun/pull/10245, but I will take care of the early exit checks there 👍.